### DISCUSSION TITLE
var_export does not handle circular references

### DISCUSSION THREAD

**Discussion by prathna124** [laravel/framework#48931] [Q&A]
Laravel: 8
Php: 8.1

We are facing an issue when caching the config with
PHP artisan config: cache. It is a circular references-related issue and we have also faced issue-related closure that occurs when using the callback function in config files as you have suggested here https://github.com/exodusanto/Laravel-Admin/blob/master/docs/fields.md#editable-option.

https://i.imgur.com/Fxcb7Pt.png
Error: var_export does not handle circular references

We have tried this but not working:-

We have tried to handle the error using Handler.php but it is not working
Also we have tried to remove the closure function from config/adminer.php but we have faced the below error:
There are problems with your administrator.php config: The permission option must be a function
Please guide me on how to solve this error when configuring the cache.

Is it possible to ignore this error when config cache or any way to handle this so error should not be reported?
Please guide me how to solve this ?

**Comment by henzeb** [comment]
Try https://packagist.org/packages/henzeb/var-export-wrapper

It should automatically wrap closures so they are exportable.

**Reply by prathna124** [reply]
We have installed this package and used it like below:-

In vendor/laravel/framework/src/Illuminate/Foundation/Console/ConfigCacheCommand.php in:-
use function Henzeb\VarExportWrapper\Support\Functions\var_export;


$configPath, '<?php return '.var_export($config).';'.PHP_EOL

faced below error:-

Exception 

  Serialization of 'Closure' is not allowed

  at vendor/henzeb/var-export-wrapper/src/VarExportable.php:35
     31▕             return;
     32▕         }
     33▕ 
     34▕         $this->object = serialize(
  ➜  35▕             $var
     36▕         );
     37▕     }
     38▕ 
     39▕     public static function __set_state(array $state)

      +3 vendor frames 
  4   [internal]:0
      Henzeb\VarExportWrapper\Support\Functions\{closure}()

      +16 vendor frames 
  21  artisan:37
      Illuminate\Foundation\Console\Kernel::handle()

Please guide me on how to solve this.
